api/v1alpha1/nicclusterpolicy_webhook.go

```diff
@@ -229,6 +226,28 @@ func (dp *DevicePluginSpec) validateSriovNetworkDevicePlugin(fldPath *field.Path
 	return allErrs
 }
 
+func validateResourceNamePrefix(resource map[string]interface{},
+	allErrs field.ErrorList, fldPath *field.Path, dp *DevicePluginSpec) (bool, field.ErrorList) {
+	resourceName := resource["resourceName"].(string)
+	if !isValidSriovNetworkDevicePluginResourceName(resourceName) {
```

Comment:
can we replace this one with isValidFQDN as well ?

Reply:
I am not sure that the resource name has the same pattern as an FQDN

Reply:
looking  at device plugin : https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/ea2e52c4ae8bf48c17ec666f605cb66fd03a3d49/pkg/utils/utils.go#L274

lets keep this as is. im not sure k8s will actually complain if it was fqdn.